ipxe/auto.go

```diff
@@ -76,27 +78,46 @@ func (h *Handler) HandlerFunc() http.HandlerFunc {
 	}
 }
 
+func customScriptFound(hw client.Discoverer, ip string) bool {
+	if hw == nil {
+		return false
+	}
+	mac := hw.GetMAC(net.ParseIP(ip))
+
+	return hw.Hardware().IPXEURL(mac) != "" || hw.Hardware().IPXEScript(mac) != ""
+}
+
 func (h *Handler) serveBootScript(ctx context.Context, w http.ResponseWriter, name string, ip string, hw client.Discoverer) {
 	span := trace.SpanFromContext(ctx)
 	span.SetAttributes(attribute.String("boots.script_name", name))
 	var script []byte
+	// check if the custom script should be used
+	if customScriptFound(hw, ip) {
+		name = "custom.ipxe"
+	}
 	switch name {
 	case "auto.ipxe":
-		// check if the custom script should be used
-		if cs, err := h.customScript(hw, ip); err == nil {
-			script = []byte(cs)
-			break
-		}
 		s, err := h.defaultScript(span, hw, ip)
 		if err != nil {
-			w.WriteHeader(http.StatusNotFound)
-			err := errors.Errorf("boot script %q not found", name)
+			w.WriteHeader(http.StatusInternalServerError)
+			err := errors.Wrap(err, fmt.Sprintf("boot script %q not found", name))
```

Comment:
maybe change this error message to be alike the custom one? e.g. `error with default ipxe script`

Reply:
sure thing, will update